Sources/CustomerCenter/Events/CustomerCenterEvent.swift

```diff
@@ -13,70 +13,127 @@
 
 import Foundation
 
-/// An event to be sent by the `RevenueCatUI` SDK.
-public enum CustomerCenterEvent: FeatureEvent {
+/// A protocol that represents a customer center event.
+public protocol CustomerCenterEventType {}
+
+extension CustomerCenterEventType {
 
-    // swiftlint:disable type_name
+    var feature: Feature { .customerCenter }
 
-    /// An identifier that represents a customer center event.
-    public typealias ID = UUID
+}
 
-    // swiftlint:enable type_name
+/// Data that represents a customer center event creation.
+public struct CustomerCenterEventCreationData {
 
-    /// An identifier that represents a paywall session.
-    public typealias SessionID = UUID
+    let id: UUID
+    let date: Date
 
-    var feature: Feature {
-        return .customerCenter
+    // swiftlint:disable:next missing_docs
+    public init(
+        id: UUID = .init(),
+        date: Date = .init()
+    ) {
+        self.id = id
+        self.date = date
     }
 
+}
+
+/// An event to be sent by the `RevenueCatUI` SDK.
+public enum CustomerCenterEvent: FeatureEvent, CustomerCenterEventType {
+
+    var eventDiscriminator: String? { "lifecycle" }
+
     /// The Customer Center was displayed.
-    case impression(CreationData, Data)
+    case impression(CustomerCenterEventCreationData, Data)
+
+}
+
+/// An event to be sent by the `RevenueCatUI` SDK.
+public enum CustomerCenterSurveyOptionChosenEvent: FeatureEvent, CustomerCenterEventType {
```

Comment:
Hmm probably missing it but why does this need to be a different class like this? could it be just part of the `CustomerCenterEvent` enum?

Reply:
The Customer Center events are more complex than the Paywall Events, and one single class can't represent all of them.

I should have linked the document with the definition but here it is https://docs.google.com/document/d/1uZ5-obuQXWDGZ-1zKh8qaVYyJOK8bDpEevyWPdXTODA/edit?usp=sharing 

As you can tell one single class can't really represent all of them. We could have created one with a bunch of nullables, but this way is cleaner. The only complication is differentiating them when deserializing them, and that's why I am using that `eventDiscriminator` to distinguish between the different types of Customer Center events when deserializing them from storage.